transport/matrix/layer.py

```diff
@@ -77,34 +60,44 @@ def __init__(self, config: Dict[str, Any]):
                                  light_client["address"])
                         continue
 
-                matrix_config = config["transport"]["matrix"]
-                matrix_config["current_server_name"] = current_server_name
+                config = config["transport"]["matrix"]
```

Comment:
are you sure this change is correct? in another PR you fixed the issue of overwriting the original `config` dict

Reply:
ups, fixed